src/Sulu/Bundle/DocumentManagerBundle/Reference/Provider/AbstractDocumentReferenceProvider.php

```diff
@@ -75,6 +75,13 @@ public function updateReferences($document, string $locale, string $context): vo
         );
 
         $structure = $document->getStructure();
+
+        if ($document->getStructureType() === null) {
+            // if there is no structureType we cannot update the references
+            // this happens for pages/articles that were published once but have been unpublished
+            return;
+        }
```

Comment:
Do we not need to still call `$referenceCollector->persistReferences();` to remove old references eventually?

Reply:
We do not need to remove them manually, when a page is unpublished they are removed automatically due to the `DocumentReferenceSubscriber` and when the refresh command is run, they are removed anyway because of the older timestamp see https://github.com/sulu/sulu/blob/2.6/src/Sulu/Bundle/ReferenceBundle/UserInterface/Command/RefreshCommand.php#L84-L88